### PR TITLE
[feature](tvf)(jni-avro)jni-avro scanner add complex data types

### PR DESCRIPTION
**Pull request by DongLiang-0**
## Proposed changes
Support avro's enum, record, union data types

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

File: regression-test/suites/external_table_p0/tvf/test_tvf_avro.groovy [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        """
+
+    qt_6 """
+           select anEnum from s3(
+            "uri" = "${s3Uri}",
+            "ACCESS_KEY" = "${ak}",
+            "SECRET_KEY" = "${sk}",
+            "REGION" = "${region}",
+            "FORMAT" = "${format}");
+        """
```
Comment (BePPPower) [inline regression-test/suites/external_table_p0/tvf/test_tvf_avro.groovy:91]: Please use `order_qt` to ensure the results are the same every time

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Review by BePPPower — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**